Simulation sheet scenario result

Uriel-Sautron opened the pull request:
closes https://github.com/osrd-project/osrd-confidential/issues/688

Uriel-Sautron committed:
front: add simulation sheet and csv export button in scenario page

Signed-off-by: Uriel Sautron <uriel.sautron@gmail.com>
Signed-off-by: theocrsb <theo_crosbie@yahoo.fr>

Uriel-Sautron committed:
front: remove driver train schedule

Signed-off-by: Uriel Sautron <uriel.sautron@gmail.com>
Signed-off-by: theocrsb <theo_crosbie@yahoo.fr>

Uriel-Sautron commented on front/public/locales/fr/operationalStudies/study.json:
```diff
@@ -19,6 +19,7 @@
   "errorHappened": "Une erreur est survenue",
   "filesAndLinks": "Documents & liens",
   "filterPlaceholder": "Rechercher…",
+  "infrastructureWithTwoPoints": "Infra : ",
```
```suggestion
  "infrastructureWithTwoPoints": "Infra: ",
```

theocrsb replied:
This space has been added because there is a difference between French and English

Uriel-Sautron commented on front/public/locales/fr/operationalStudies/study.json:
```diff
@@ -35,8 +36,10 @@
   "scenarioDeletedMulti_one": "{{count}} scénario a été supprimé.",
   "scenarioDeletedMulti_other": "{{count}} scénarios ont été supprimés.",
   "scenariosList": "Scénarios",
+  "scenarioWithTwoPoints": "Scénario : ",
```
```suggestion
  "scenarioWithTwoPoints": "Scénario: ",
```

theocrsb replied:
Same here

Akctarus commented on front/src/applications/stdcm/views/StdcmResultsV2.tsx:
```diff
@@ -74,6 +74,7 @@ const StcdmResultsV2 = ({ mapCanvas, stdcmV2Results, pathProperties }: StcdmResu
             <PDFDownloadLink
               document={
                 <SimulationReportSheet
+                  simulationType="stdcm"
```
You can remove it

theocrsb replied:
Done

Akctarus commented on front/src/applications/stdcmV2/components/StdcmResults/StdcmResults.tsx:
```diff
@@ -98,6 +98,7 @@ const StcdmResults = ({
                   <PDFDownloadLink
                     document={
                       <SimulationReportSheet
+                        simulationType="stdcm"
```
You can remove it

theocrsb replied:
Done

Akctarus commented on front/src/applications/stdcm/types.ts:
```diff
@@ -23,6 +23,7 @@ export type StdcmV2SuccessResponse = Omit<
 };
 
 export type SimulationReportSheetProps = {
+  simulationType?: 'stdcm' | 'scenario';
```
you can remove it

theocrsb replied:
Done

Akctarus commented on front/src/applications/operationalStudies/views/SimulationResults.tsx:
```diff
@@ -6,6 +6,7 @@ import { useTranslation } from 'react-i18next';
 
 import type { SimulationResultsData } from 'applications/operationalStudies/types';
 import type { Conflict } from 'common/api/osrdEditoastApi';
+import type { ScenarioResponse } from 'common/api/osrdEditoastApi';
```
```suggestion
import type { Conflict, ScenarioResponse } from 'common/api/osrdEditoastApi';
```

theocrsb replied:
Done

Akctarus commented on front/src/modules/simulationResult/components/SimulationResultsMap/SimulationResultsMap.tsx:
```diff
@@ -150,6 +159,25 @@ const SimulationResultMap = ({ geometry, trainSimulation }: SimulationResultMapP
+        console.error('Error capturing map:', error);
+      }
+    }
+  };
```
Since the same function is used in 2 files, it could be good to define it once in utils file then import it

theocrsb replied:
Done

Akctarus commented on front/src/modules/simulationResult/SimulationResultExport/SimulationReportSheetScenario.tsx:
```diff
@@ -0,0 +1,328 @@
+          <View style={styles.header.numericInfo}>
+            <Text style={styles.header.number}>
+              {t('operationalStudies/study:scenarioWithTwoPoints')}
+              {scenarioResponse?.name}
```
the suspension points extend beyond the square when title is too long

theocrsb replied:
Done

Akctarus commented on front/src/modules/simulationResult/SimulationResultExport/SimulationReportSheetScenario.tsx:
```diff
@@ -0,0 +1,328 @@
+            <View style={styles.convoyAndRoute.convoyInfo}>
+              <View style={styles.convoyAndRoute.convoyInfoBox1}>
+                <Text style={styles.convoyAndRoute.convoyInfoTitles}>{t('speedLimitByTag')}</Text>
+                <Text style={styles.convoyAndRoute.convoyInfoData}>{speedLimitByTag || '-'}</Text>
```
"speedLimitByTag" string is always displayed

theocrsb replied:
Done

Akctarus commented on front/src/modules/simulationResult/SimulationResultExport/SimulationReportSheetScenario.tsx:
```diff
@@ -0,0 +1,328 @@
+                    <TD>{t('startStop')}</TD>
+                  </View>
+                </TH>
+                {operationalPointsList.map((step, index) => {
```
results are shifted slightly to the left:
<img width="970" alt="Capture d’écran 2024-10-21 à 17 08 11" src="https://github.com/user-attachments/assets/77d9b132-f5f5-4cd5-bc88-3aabc8882252">

theocrsb replied:
Done

Akctarus commented on front/src/modules/simulationResult/SimulationResultExport/SimulationReportSheetScenario.tsx:
```diff
@@ -0,0 +1,328 @@
+                          : ''}
+                      </TD>
+                    </View>
+                    <View style={styles.simulation.passageWidth}>
```
Stops are weirdly displayed:
<img width="1390" alt="Capture d’écran 2024-10-21 à 17 09 51" src="https://github.com/user-attachments/assets/a44d3e17-e4cf-4108-a6fd-5c123ff6556c">

theocrsb replied:
Done

Akctarus reviewed (changes requested):
Tested, left few comments

SharglutDev commented on front/src/modules/trainschedule/components/DriverTrainSchedule/DriverTrainScheduleHeader.tsx:
```diff
@@ -1,124 +0,0 @@
-import { BaseOrEco, BASE_OR_ECO_OPTIONS, type BaseOrEcoType } from './consts';
-import exportTrainCSV from './exportDriverScheduleCSV';
-import type { OperationalPointWithTimeAndSpeed } from './types';
-import { isEco, massWithOneDecimal } from './utils';
```
You can remove the `isEco` function

theocrsb replied:
Done

theocrsb replied:
Done

SharglutDev commented on front/src/modules/trainschedule/components/DriverTrainSchedule/DriverTrainSchedule.tsx:
```diff
@@ -1,76 +0,0 @@
-import { BaseOrEco, type BaseOrEcoType } from './consts';
-import DriverTrainScheduleHeader from './DriverTrainScheduleHeader';
-import DriverTrainScheduleStopList from './DriverTrainScheduleStopList';
-import type { OperationalPointWithTimeAndSpeed } from './types';
```
This type should be moved in `operationalStudies/types.ts`

theocrsb replied:
Done

theocrsb replied:
Done

SharglutDev commented on front/src/modules/trainschedule/components/DriverTrainSchedule/DriverTrainScheduleStop.tsx:
```diff
@@ -1,114 +0,0 @@
-import type { PositionSpeedTime, SpeedRanges } from 'reducers/simulationResults/types';
-
-import type { OperationalPointWithTimeAndSpeed } from './types';
-import { getActualSpeed, getActualVmax, getAverageSpeed, getTime } from './utils';
```
`getActualSpeed`, `getAverageSpeed` and `getTime` can be removed (and maybe some functions using them)

theocrsb replied:
Done

SharglutDev commented on front/src/styles/scss/_fonts.scss:
```diff
@@ -5,6 +5,12 @@
   font-weight: 400;
 }
 
+@font-face {
+  font-family: 'IBM Plex Sans';
+  src: inline('assets/newFonts/IBM-Plex-Sans/IBMPlexSans-Medium.ttf') format('truetype');
+  font-weight: 500;
+}
```
Seems the font doesn't work

theocrsb replied:
Done

SharglutDev commented on front/src/styles/scss/applications/operationalStudies/_simulationresults.scss:
```diff
@@ -48,6 +48,17 @@
+    padding-top: 1rem;
+    button {
+      font-weight: 500;
+      background-color: rgba(235, 235, 234, 0.4);
```
For consistency, we should add a var for this color in hex. We could name it `grey10O40: #ebebea66` (O for opacity)

theocrsb replied:
I just add a comment between bg color

Akctarus commented on front/src/modules/simulationResult/SimulationResultExport/SimulationReportSheetScenario.tsx:
```diff
@@ -0,0 +1,321 @@
+                          <TD style={styles.convoyAndRoute.stopTableStartColumn}>
+                            {isFirstStep ? secToHoursString(step.time, { withSeconds: true }) : ''}
+                          </TD>
+                        </View>
```
The departure and arrival times are offset by minus 1 hour from the actual time. Also, to match the STDCM simulation sheet, I don't think the seconds should be displayed.

Same for the same for the PR table, all the displayed times are missing 1 hour.

Uriel-Sautron replied:
By entering the current date, we obtain the correct times. 
It must have been the management of winter hours that wasn't happening.

Akctarus commented on front/src/modules/simulationResult/SimulationResultExport/SimulationReportSheetScenario.tsx:
Passage stop are not displayed on tables (you can check the STDCM sheet to to retrieve the logic)

Uriel-Sautron replied:
Done

theocrsb committed:
front: fix font problem
  - Font files have been moved from src/asset to plublic. In public, paths are stable, which facilitates their use in CSS files.

Signed-off-by: theocrsb <theo_crosbie@yahoo.fr>

SharglutDev commented on front/src/styles/scss/_fonts.scss:
Could you add a description to the font commit (3rd) explaining why you had to make these changes

SharglutDev commented on front/src/applications/stdcm/components/StdcmResults/SimulationReportStyleSheet.ts:
```diff
@@ -107,6 +98,15 @@ const styles = {
       marginTop: '10',
       marginBottom: '3',
     },
+    cardContent: {
```
Not very familiar with the sheet, do we use this style somewhere ?

Uriel-Sautron replied:
Yes is used in SimulationReportSheetScenario.tsx
`<Text style={styles.header.cardContent}>{trainName}</Text>`

SharglutDev commented on front/src/modules/simulationResult/SimulationResultExport/types.ts:
```diff
@@ -0,0 +1,9 @@
+import type { RollingStockWithLiveries } from 'common/api/osrdEditoastApi';
+
+export type SimulationSheetData = {
+  trainName: string | undefined;
+  rollingStock: RollingStockWithLiveries;
+  speedLimitByTag: string | undefined | null;
+  departure_time: string;
+  creationDate: Date;
+};
```
```suggestion
export type SimulationSheetData = {
  trainName?: string;
  rollingStock: RollingStockWithLiveries;
  speedLimitByTag?: string | null;
  departure_time: string;
  creationDate: Date;
};
```

Uriel-Sautron replied:
Done

SharglutDev commented on front/src/modules/simulationResult/SimulationResultExport/utils.ts:
In the `formatOperationalPoints` function, we return the "base" operational points but it seems we never use them anywhere. We can probably remove this logic.

Uriel-Sautron replied:
Done

SharglutDev reviewed (approved):
Lgtm and tested
